api/app/Generators/PoolCandidateCsvGenerator.php

```diff
@@ -269,4 +261,35 @@ private function generatePoolHeaders()
                 }
             });
     }
+
+    private function buildQuery()
+    {
+        $query = PoolCandidate::with([
+            'generalQuestionResponses' => ['generalQuestion'],
+            'screeningQuestionResponses' => ['screeningQuestion'],
+            'educationRequirementExperiences',
+            'user' => [
+                'department',
+                'currentClassification',
+                'userSkills' => ['skill', 'experiences' => ['skills']],
+                'awardExperiences' => ['userSkills' => ['skill']],
+                'communityExperiences' => ['userSkills' => ['skill']],
+                'educationExperiences' => ['userSkills' => ['skill']],
+                'personalExperiences' => ['userSkills' => ['skill']],
+                'workExperiences' => ['userSkills' => ['skill']],
+            ],
+        ]);
+
+        $query = $this->applyFilters($query, PoolCandidate::class, [
```

Comment:
Feel free to ignore this if you don't agree, but it looks weird to me to reassign $query here.  At first glance it seems like the original value from L267 is being overwritten.  I think applyFilters is just returning the same Builder, right?

Reply:
Same comment for api/app/Generators/UserCsvGenerator.php L160, api/app/GraphQL/Mutations/DownloadPoolCandidatesCsv.php L37, and api/app/GraphQL/Mutations/DownloadUsersCsv.php L38.

Reply:
Was this what you had in mind?

[remove reassignments](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11300/commits/bc77db820f0125b28dcce1a7d2860d25bf83504f)

Reply:
Yeah, exactly.  You might still want to return the Builder from the function for chaining,

Reply:
If you do keep the ": void" signature you'll need to get rid of the `return $query;` on L72.  It's causing an HTTP500 right now.

Reply:
Yeah, even though we are not chaining I like keeping that functionality since it is a common pattern for query builers.

[remove need to pass class in](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11300/commits/a2e00b2c8fa2d47bfe7412318a806a233ce7ec2c)

Reply:
Yeah, I like this.  :+1: